staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -618,22 +679,49 @@ func (p *patcher) patchResource(ctx context.Context, scope *RequestScope) (runti
 // <originalMap> and stores the result in <objToUpdate>.
 // NOTE: <objToUpdate> must be a versioned object.
 func applyPatchToObject(
+	requestContext context.Context,
 	defaulter runtime.ObjectDefaulter,
 	originalMap map[string]interface{},
 	patchMap map[string]interface{},
 	objToUpdate runtime.Object,
 	schemaReferenceObj runtime.Object,
+	strictErrs []error,
+	validationDirective string,
 ) error {
 	patchedObjMap, err := strategicpatch.StrategicMergeMapPatch(originalMap, patchMap, schemaReferenceObj)
 	if err != nil {
 		return interpretStrategicMergePatchError(err)
 	}
 
 	// Rather than serialize the patched map to JSON, then decode it to an object, we go directly from a map to an object
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(patchedObjMap, objToUpdate); err != nil {
-		return errors.NewInvalid(schema.GroupKind{}, "", field.ErrorList{
-			field.Invalid(field.NewPath("patch"), fmt.Sprintf("%+v", patchMap), err.Error()),
-		})
+	converter := runtime.DefaultUnstructuredConverter
+	if err := converter.FromUnstructuredWithValidation(patchedObjMap, objToUpdate, validationDirective); err != nil {
+		if !runtime.IsStrictDecodingError(err) {
```

Comment:
similar comment as above about restructuring this to be more concise and avoiding unreachable blocks entirely

Reply:
similar comment to https://github.com/kubernetes/kubernetes/pull/105916/files?w=1#r746119701 about there being bugs when len(strictErrs) > 0 and FromUnstructuredWithValidation doesn't return an error or returns a non-strict error